spec/support/matchers/email_matchers.rb

```diff
@@ -0,0 +1,22 @@
+RSpec::Matchers.define :receive_email do
+  match do |address|
+    emails =
+      ActionMailer::Base.deliveries.select { |e| e.to.include?(address) }
+
+    emails.select! { |e| e.subject.match(@subject) } if @subject
+    emails.select! { |e| e.text_part.body.match(@body) } if @body
+    emails.any?
+  end
+
+  chain :with_subject do |subject|
```

Comment:
With <a href="https://github.com/rspec/rspec-expectations/pull/644">this</a> change in RSpec 3.2+ we can shorten the `chain`'s to something like this:

```ruby
chain :with_subject, :subject
chain :with_body, :body
chain :and_body, :body
```